src/main/java/slimeknights/tconstruct/plugin/jei/JEIPlugin.java

```diff
@@ -0,0 +1,62 @@
+package slimeknights.tconstruct.plugin.jei;
+
+import mezz.jei.api.IModPlugin;
+import mezz.jei.api.JeiPlugin;
+import mezz.jei.api.helpers.IGuiHelper;
+import mezz.jei.api.helpers.IJeiHelpers;
+import mezz.jei.api.registration.IRecipeCatalystRegistration;
+import mezz.jei.api.registration.IRecipeCategoryRegistration;
+import mezz.jei.api.registration.IRecipeRegistration;
+import net.minecraft.client.Minecraft;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.util.ResourceLocation;
+import slimeknights.tconstruct.TConstruct;
+import slimeknights.tconstruct.library.recipe.RecipeTypes;
+import slimeknights.tconstruct.plugin.jei.casting.CastingBasinCategory;
+import slimeknights.tconstruct.plugin.jei.casting.CastingTableCategory;
+import slimeknights.tconstruct.smeltery.TinkerSmeltery;
+import slimeknights.tconstruct.smeltery.recipe.ICastingInventory;
+
+import javax.annotation.Nullable;
+import java.util.Collection;
+
+@JeiPlugin
+public class JEIPlugin implements IModPlugin {
+
+  @Nullable
+  private CastingBasinCategory castingBasinCategory;
+
+  @Nullable
+  CastingTableCategory castingTableCategory;
+
+  @Override
+  public ResourceLocation getPluginUid() {
+    return TConstructRecipeCategoryUid.pluginUid;
+  }
+
+  @Override
+  public void registerCategories(IRecipeCategoryRegistration registry) {
+    final IJeiHelpers jeiHelpers = registry.getJeiHelpers();
+    final IGuiHelper guiHelper = registry.getJeiHelpers().getGuiHelper();
+    castingBasinCategory = new CastingBasinCategory(guiHelper);
+    castingTableCategory = new CastingTableCategory(guiHelper);
+    registry.addRecipeCategories(castingBasinCategory);
+    registry.addRecipeCategories(castingTableCategory);
+  }
+
+  @Override
+  public void registerRecipes(IRecipeRegistration register) {
+    Collection<IRecipe<ICastingInventory>> castingBasinRecipes = Minecraft.getInstance().world.getRecipeManager().getRecipes(RecipeTypes.CASTING_BASIN).values();
```

Comment:
Look into `RecipeUtil.getRecipes`, it will let you apply a filter class, such as if you only want `CastingBasinRecipe` to be added. Or just get the generic proper type list if you wish (saves potential unhandled recipes in JEI):
```java
List<AbstractCastingRecipe> = RecipeUtil.getRecipes(recipeManager, RecipeTypes.CASTING_BASIN, AbstractCastingRecipe.class);
```